lib/sockets-monitor.js

```diff
@@ -0,0 +1,59 @@
+'use strict';
+var child = require('child_process');
+
+
+module.exports = SocketsMonitor;
+
+/**
+ * Monitor class for performing monitoring on:
+ * - number of **open** sockets
+ * - number of **pending** sockets
+ * - number of **open** files
+ * @class
+ */
+function SocketsMonitor (monitor, prefix, interval) {
+  this.monitor = monitor;
+  this.prefix = prefix;
+  this.interval = interval;
+}
+
+/**
+ * Start capturing of the sockets stats data.
+ */
+SocketsMonitor.prototype.start = function () {
+  if (this.intervalId) { return; }
+  this.intervalId = setInterval(this._captureSocketStats.bind(this), this.interval);
+};
```

Comment:
nit: Perhaps wrap the interval set in the if? Like so:
```js
if (!this.intervalId) {
  this.intervalId = setInterval(this._captureSocketStats.bind(this), this.interval);
}
```
or change the wrapping below to short circuit?